### ISSUE TITLE
[Bug]: VSCode: `noConfigSearch` not working as expected

### ISSUE THREAD

**Issue by Nightblade** [streetsidesoftware/vscode-spell-checker#2520]
### Tool or Library

cspell

### Version

v2.16.0 (VSCode extension)

### Supporting Library

Not sure

### OS

Windows

### OS Version

Win10 Pro x64 21H2

### Description

In the VSCode CSpell extension, the `noConfigSearch` option isn't working as expected -- CSpell still finds and uses config files that are "local" to files _opened outside the current VSCode workspace folder_.

### Steps to Reproduce

1. Create a new folder called `primary`
2. Create a `primary\README.MD` file
3. Create a minimal `primary\cspell.json` file, contents:  `{ "version": "0.2" }`
5. Make a copy of the `primary` folder called `secondary`
6. Ensure your layout looks like this:
```
root
├─┬ primary
│ ├── README.md
│ └── cspell.json
├─┬ secondary
│ ├── README.md
│ └── cspell.json
```
7. Open VSCode
8. (Install the CSpell extension as needed)
9. File -> Open Folder -> `primary`
10. File -> Preferences -> settings -> `User` 
11. Enable `noConfigSearch` in the CSpell section (cSpell.noConfigSearch)
12. File -> Open File -> `secondary\README.md`
13. Open the `Spell Checker Preferences` window
14. Click the `FILE` tab
15.  Make sure the "File" drop-down shows `\secondary\README.md`
16. `Config Files` should now show `secondary/cspell.json`

### Expected Behavior

At step 16 above,  `Config Files` should show `primary/cspell.json` only.

### cspell.json

```js
{ "version": "0.2" }
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
